iOSDFULibrary/Classes/Implementation/LegacyDFU/Characteristics/DFUControlPoint.swift

```diff
@@ -221,7 +221,11 @@ internal struct PacketReceiptNotification {
         self.report  = report
         
         // Get the peripheral object.
+        #if swift(>=5.5)
+        guard let service = characteristic.service, let peripheral = service.peripheral else { return }
```

Comment:
Here, when Swift 5.5. is used, the code below has access to `services`, while older versions will not. This may lead to uncough compilation errors when older Xcode is used. I would rather have it like:
```swift
guard let peripheral = characteristic?.service.peripheral else { return }
```
if that's possible.